hvplot/converter.py

```diff
@@ -1485,7 +1486,9 @@ def _process_style(self, kwds, plot_opts):
         if 'color' in style_opts:
             color = style_opts['color']
         elif not isinstance(cmap, dict):
-            if cmap and any(c in cmap for c in categories):
+            if (isinstance(cmap, str) or _mpl_cmap(cmap)) and any(
+                c in getattr(cmap, 'name', cmap) for c in categorical_cmaps
```

Comment:
This is slick! It's a tad hard to comprehend so I'd leave a comment to help out future contributors on what this does, e.g.

# Checks if any of the categorical cmaps matches cmap; uses any() instead of `cmap in categorical_cmaps` to handle reversed colormaps (suffixed with `_r`). If cmap is LinearSegmentedColormap, get the name attr, else return the str typed cmap.

Alternatively, get rid of the `any` and make it more intuitive by checking if `cmap in categorical_cmap` or `cmap in f'{categorical_cmap}_r'` 

Lastly, to make it more readable, i.e. sound like choppy English, I would rename:
1. `c` to `categorical_cmap`
2. `_mpl_cmap` to `is_mpl_cmap` (for consistency too, e.g. is_xarray)

Reply:
Done. I left the usage of `any` 'cos I thought it more standard and concise :)